internal/bminventory/inventory.go

```diff
@@ -3882,58 +3884,37 @@ func (b *bareMetalInventory) RegisterInfraEnvInternal(
 				errMsg = fmt.Sprintf("%s. Error: %s", errMsg, errString)
 			}
 			log.Errorf(errMsg)
+			tx.Rollback()
 			eventgen.SendInfraEnvRegistrationFailedEvent(ctx, b.eventsHandler, id, errString)
 		}
 	}()
 
 	params = b.setDefaultRegisterInfraEnvParams(ctx, params)
 
-	if err = auth.ValidateAccessToCPUArchitecture(ctx, b.authzHandler, params.InfraenvCreateParams.CPUArchitecture); err != nil {
-		return nil, err
-	}
+	var cluster *common.Cluster
+	clusterId := params.InfraenvCreateParams.ClusterID
+	if clusterId != nil {
+		cluster, err = common.GetClusterFromDB(b.db, *clusterId, common.SkipEagerLoading)
+		if err != nil {
+			err = errors.Errorf("Cluster ID %s does not exists", clusterId.String())
```

Comment:
Should this continue to be a `BadRequest` response as it was in https://github.com/openshift/assisted-service/pull/3791/files#diff-747a429a0e12307ef65d754c83b6898ce3858b621005e1e7af6e420da1f1657cL4068

Reply:
Right, good catch! Modified it back

Reply:
Why isn't it NotFound?

Reply:
I suppose it's because the cluster ID is provided as input vs being the resource you're operating on.
That said I was just comparing to the previous implementation.

I guess NotFound would work if the cluster actually wasn't found, but to be super pedantic, I'd expect BadRequest if the ID was something other than a uuid and I'd expect InternalServerError if some other database error happened (connection error, perhaps).

Reply:
Agree with Nick

Reply:
Went ahead and took a closer look

If the ID is not a UUID then the error is caught earlier on, but it doesn't return a `BadRequest` (at least not from curling the API):
```
$ curl ${ASSISTED_SERVICE_URL}/api/assisted-install/v2/infra-envs -X POST -H "Content-Type: application/json" --data '{"name":"test2","cluster_id": "test"}'

{"code":601,"message":"cluster_id in body must be of type uuid: \"test\""}
```
We might be able to rule out the cluster_id not being type UUID as an error.

This leaves:
1. Cluster_id not found --> `NotFound` error
2. DB error --> `InternalServerError`

I've made changes to reflect this. How does it look to you both?